dom.go

```diff
@@ -44,6 +50,11 @@ type Component interface {
 	// Context returns the components context, which is used internally by
 	// Vecty in order to store the previous component render for diffing.
 	Context() *Core
+
+	// ComponentOrHTML is satisfied by vecty.Core
+	ComponentOrHTML
+	// MarkupOrChild is satisfied by vecty.Core
+	MarkupOrChild
```

Comment:
We should instead just add the private methods here:

```Go
    isComponentOrHTML()
    isMarkupOrChild()
```

Then these will not show up in our godoc and instead it will show:

```
type Component interface {
    ...exported methods...
    // contains filtered or unexported methods
}
```

For an example of this, see https://golang.org/pkg/reflect/#Type